### ISSUE TITLE
Rivers get in the way

### ISSUE THREAD

**Issue by Inglonias** [TGWeaver/CDDA-Sky-Islands#49]
Pretty simple issue, probably not so simple solution:

On my last character's first expedition there was a river between myself and the extraction point. There was no way for me to know where the nearest bridge was. I gave up on that character right then and there.

If it is at all possible to ensure that a walkable path exists that won't take longer than the time limit to walk on between myself and the extraction point, that would be great. This is probably not possible, or if it is, not feasible.

A band aid solution might be to lower the occurrence of rivers in the overworld.

**Comment by TGWeaver** [comment]
Yeah, no overmap pathfinding is available for the code, possibly at all, and you need ways past rivers the most when you can least afford them. There is a way to just turn them off entirely, and I may include that as a file in the next release (since it can't be toggled while in-game, only via files).

**Comment by Inglonias** [comment]
That might be the best solution.

It's worth noting that a failed expedition or two was not nearly as catastrophic as I had feared, so this is not quite as urgent. If you fail early on, you probably aren't losing anything irreplaceable, and if you fail later, you can use the warp shards spell to come home.

**Comment by DenCheesecake** [comment]
I've played hundreds of raids in Sky Island and never had a bad spawn on an island. You must have been immensely unlucky. I bumped into a few rivers, sure, but if you follow the river you will find a bridge eventually. It has never been a problem in my case. Just sharing my feedback

**Comment by Inglonias** [comment]
It is less of an issue on longer expeditions, but in 0.3.0 you are limited to two hour expeditions at the start. Two hours is not a lot of time to find your way around a river and depending on how far that bridge is, may be too far to travel in a short time.

**Comment by Khrazlon** [comment]
Would it be possible to have the exits spawn near to roads or trails, to reduce the chance of them being on islands or unreachable?

**Comment by Coach-mdl** [comment]
Perhaps a spell that helps the player float in water could help. The HỎI could teach it. It doesn't solve the river issue, but it gives the player a means to cross if all else fails. 

The main issue isn't inherently the river itself but how feasible it is to cross in the time you have. If one wanted to make a boat, they could, but it would have to be foldable and made pre-raid and is not early game accessible. Flotation vests could help too but that's extra storage taken up to help with a scenario that doesn't happen consistently enough to warrant carrying it. We just need to have a way across without leaving everything behind to swim.
